WooCommerce/Classes/ViewRelated/Products/Variations/Bulk Update/BulkUpdateViewModel.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+import Yosemite
+import protocol Storage.StorageManagerType
+
+/// View Model logic for the Bulk Variations Update
+final class BulkUpdateViewModel {
+
+    /// Represents possible states for syncing product variations.
+    enum SyncState: Equatable {
+        case syncing
+        case syncedResults
+        case syncingError
+        case notStarted
+    }
+
+    private let storageManager: StorageManagerType
+    private let storesManager: StoresManager
+    private let siteID: Int64
+    private let productID: Int64
+
+    /// The state of synching all variations
+    private(set) var syncState: SyncState {
+        didSet {
+            observeSyncState?(syncState)
+        }
+    }
+
+    /// Called when syncing state changes
+    var observeSyncState: ((SyncState) -> Void)?
+
+    init(siteID: Int64,
+         productID: Int64,
+         storageManager: StorageManagerType = ServiceLocator.storageManager,
+         storesManager: StoresManager = ServiceLocator.stores) {
+        self.siteID = siteID
+        self.productID = productID
+        self.storageManager = storageManager
+        self.storesManager = storesManager
+        syncState = .notStarted
+    }
+
+    /// This is the main activation method for this view model.
+    ///
+    /// This should only be called when the corresponding view was loaded.
+    ///
+    func activate() {
+        synchAllProductVariations()
+    }
+
+    /// Start synching all product variations.
+    ///
+    func synchAllProductVariations() {
+        syncState = .syncing
+
+        let numberOfObjects = Constants.numberOfObjects
+        // There is a limitof 100 objects for bulk update API, so we fetch 101 so see if user has more that 100.
+        let pageNumber = Constants.pageNumber
+        let action = ProductVariationAction
+            .synchronizeProductVariations(siteID: siteID, productID: productID, pageNumber: pageNumber, pageSize: numberOfObjects) { [weak self] error in
+                guard let self = self else {
+                    return
+                }
+
+                if let error = error {
+                    self.syncState = .syncingError
+
+                    DDLogError("⛔️ Error synchronizing product variations: \(error)")
+                } else {
+                    self.syncState = .syncedResults
+                }
+            }
+
+        storesManager.dispatch(action)
+    }
```

Comment:
I'm failing to see how this fetches all variations. I think this fetches the first `100`.
I think what we need is to have a Yosemite action that fetches all variations and abstracts this logic from the view model. what do you think?

Reply:
Since we have a limit on the bulk update API call of 100 we will trim the bulk update network request to 100 objects, so I am fetching here only what is needed. 
I have updated the method name `synchAllProductVariations`->`synchProductVariations` and added better documentation to indicate that we are not fetching all variations.

As part of this issue I will add logic to also fetch one more page in order to show the user a warning of she has >100 variations. (I added an extra task for this)